ethylene-core/src/main/java/com/github/steanky/ethylene/core/collection/AbstractConfigList.java

```diff
@@ -26,21 +28,54 @@ public abstract class AbstractConfigList extends AbstractList<ConfigElement> imp
 
     /**
      * Construct a new AbstractConfigList using the provided list to store its elements.
+     *
      * @param list the backing list
      * @throws NullPointerException if list is null
      */
     protected AbstractConfigList(@NotNull List<ConfigElement> list) {
         this.list = Objects.requireNonNull(list);
     }
 
+    /**
+     * This helper method can be used to construct a list with the same elements as a collection. If the collection
+     * contains any null values, a {@link NullPointerException} will be thrown. Furthermore, each value will be tested
+     * against the provided {@link Predicate}. If it returns false, an {@link IllegalArgumentException} will be thrown.
+     *
+     * @param collection     the collection whose elements will be added to the new list
+     * @param listSupplier   the supplier used to create the new list from the size of the original collection
+     * @param valuePredicate the predicate to use to validate each element against some condition
```

Comment:
I feel like the valuePredicate would not generally be useful, maybe add an overload for unused -> true

Reply:
i change my mind, its protected